src/App.tsx

```diff
@@ -8,7 +8,105 @@
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { getUser } from './api';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { TodoInfo } from './types/TodoInfo';
+
 export const App: React.FC = () => {
+  const [selectedTodoId, setSelectedTodoId] = useState<number | null>(null);
+  const [isLoad, setIsLoad] = useState(false);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoadTodoModal, setIsLoadTodoModal] = useState(false);
+  const [todoInfo, setTodoInfo] = useState<TodoInfo | null>(null);
+  const [filter, setFilter] = useState<string>('all');
+  const [query, setQuery] = useState<string>('');
+
+  function handleFilter(e: React.ChangeEvent<HTMLSelectElement>) {
+    setFilter(e.target.value);
+  }
+
+  function handleQuery(e: React.ChangeEvent<HTMLInputElement>) {
+    setQuery(e.target.value);
+  }
+
+  function handleDeleteQuery() {
+    setQuery('');
+  }
+
+  function handleSelectTodo(id: number) {
+    setSelectedTodoId(id);
+  }
+
+  function handleCloseTodo() {
+    setSelectedTodoId(null);
+  }
+
+  useEffect(() => {
+    const fetchTodos = async () => {
+      setIsLoad(true);
+      try {
+        const todosFromServer = (await getTodos()).filter(todo =>
+          todo.title.toLowerCase().includes(query.toLowerCase()),
+        );
+
+        switch (filter) {
+          case 'all':
+            setTodos(todosFromServer);
+            break;
+          case 'active':
+            const activeFirst = [...todosFromServer].filter(a => !a.completed);
+
+            setTodos(activeFirst);
+            break;
+          case 'completed':
+            const completedFirst = [...todosFromServer].filter(
+              a => a.completed,
+            );
+
+            setTodos(completedFirst);
+            break;
+          default:
+            break;
+        }
+      } finally {
+        setIsLoad(false);
+      }
+    };
+
+    fetchTodos();
+  }, [filter, query]);
+
+  useEffect(() => {
+    if (selectedTodoId === null) {
+      return;
+    }
+
+    const fetchUsers = async () => {
+      setIsLoadTodoModal(true);
```

Comment:
Consider renaming `setIsLoadTodoModal` to `setIsLoadingTodoModal` for better readability and to follow common naming conventions for boolean variables.